### PR TITLE
fix(chips): Fix choice-chips leading icon being hidden

### LINKED ISSUE #2728: Choice chips hide the leading icon when selected
Choice chips hide the leading icon when selected.

Example: https://codepen.io/patty/pen/QrmzaR

### PR DESCRIPTION
**Pull request by EstebanG23**
Fixes #2728 

Add a boolean to not remove the leading icon when selected chip does not contain a checkmark.
Change the color of the leading icon when selected. Change the tests to ensure it is checking the right chips. Add test for selected chip with leading icon and no checkmark, to not
remove the leading icon.

### CODE REVIEW

File: packages/mdc-chips/chip/adapter.js [hunk 1]
```diff
@@ -47,6 +47,12 @@ class MDCChipAdapter {
    */
   hasClass(className) {}
 
+  /**
+   * Returns true if contains the checkmark class.
+   * @return {boolean}
+   */
+  hasCheckmark() {}
```
Comment (bonniezhou) [inline packages/mdc-chips/chip/adapter.js]: Just realized...if we're restricting the opacity changes to filter chips in CSS, then we don't need to check for checkmarks in the JS anymore :D

File: packages/mdc-chips/chip/mdc-chip.scss [hunk 2]
```diff
@@ -115,7 +115,7 @@
 
 // Add leading checkmark to filter chips with a leading icon
 
-.mdc-chip__icon--leading {
+.mdc-chip-set--filter .mdc-chip__icon--leading {
```
[the remark this answers is not in the document]
Comment (EstebanG23) [reply packages/mdc-chips/chip/mdc-chip.scss]: Done!

File: packages/mdc-chips/chip/adapter.js (hunk 1 above)
Comment (EstebanG23) [reply packages/mdc-chips/chip/adapter.js]: Done!

File: packages/mdc-chips/chip/foundation.js [hunk 3]
```diff
@@ -161,7 +161,7 @@ class MDCChipFoundation extends MDCFoundation {
         this.adapter_.hasClass(cssClasses.SELECTED)) {
       this.adapter_.addClassToLeadingIcon(cssClasses.HIDDEN_LEADING_ICON);
     } else if (this.adapter_.eventTargetHasClass(/** @type {!EventTarget} */ (evt.target), cssClasses.CHECKMARK) &&
-               !this.adapter_.hasClass(cssClasses.SELECTED)) {
+                !this.adapter_.hasClass(cssClasses.SELECTED)) {
```
Comment (bonniezhou) [inline packages/mdc-chips/chip/foundation.js]: Move to one line :)

**Pull request merged**